spec/websocket_spec.lua

```diff
@@ -243,23 +243,62 @@ describe("http.websocket module two sided tests", function()
 		assert.truthy(cq:empty())
 	end)
 	local function send_receive_test(name, data, data_type)
-		it(name, function()
-			data_type = data_type or "text"
-			local cq = cqueues.new()
-			local c, s = new_pair()
-			cq:wrap(function()
-				assert(c:send(data, data_type))
-				assert.same({data, data_type}, {assert(c:receive())})
-				assert(c:close())
+		local function do_test(bad_connection)
+			local bname = name
+			if bad_connection then
+				bname = bname .. ", even if the connection is really bad"
+			end
+			it(bname, function()
+				data_type = data_type or "text"
+				if bad_connection then
+					local real_xwrite
+					local fragments = 100
+					local delay = 1 / fragments -- Aim for 1s.
+					real_xwrite = cs.interpose("xwrite", function(self, str, mode, timeout)
```

Comment:
You don't need to interpose the whole library: you should be able to overwrite just the specific instance:
```lua
local c, s = new_pair()
local old_xwrite = s.socket.xwrite
s.socket.xwrite = function(.......
```

Reply:
It just seemed like the intended way to do it, but sure, I could do that.

Reply:
Ah never mind, I remember now why I did it like this. CQS sockets are userdata, you can't just override their methods like that.

Reply:
I'm open to alternatives.